src/engine/contactwriter.cpp

```diff
@@ -2530,6 +2546,64 @@ static ContactWriter::DetailList allSingularDetails()
     return details;
 }
 
+static QContactDetail supportedDetailInstance(QContactDetail::DetailType type)
+{
+    switch (type) {
```

Comment:
Ok, but you probably don't need to do the return statements from the switch - if you just validate the type, you can do the short-form return statement.